buildSrc/src/main/groovy/io/micronaut/guides/GuideProjectGenerator.groovy

```diff
@@ -103,65 +119,38 @@ class GuideProjectGenerator implements AutoCloseable {
                 throw new Exception("Guide metadata " + configFile + 'does not validate the JSON Schema\n' + assertions)
             }
         }
 
-
-
-
-        List<Category> categories = []
-        for (String c : config.categories) {
-            Category cat = Category.values().find { it.toString() == c }
-            if (cat) {
-                categories << cat
-            } else if (publish && !cat) {
-                throw new GradleException("$configFile.parentFile.name metadata.category=$config.category does not exist in Category enum")
-            }
-        }
-
-        Optional.ofNullable(new GuideMetadata(
-                asciidoctor: publish ? dir.name + '.adoc' : null,
-                slug: dir.name,
-                title: config.title,
-                intro: config.intro,
-                authors: config.authors,
-                tags: config.tags,
-                cloud: config.cloud != null ? Cloud.valueOf(config.cloud.toUpperCase()) : null,
-                categories: categories,
-                publicationDate: publish ? LocalDate.parse(config.publicationDate) : null,
-                publish: publish,
-                base: config.base,
-                languages: config.languages ?: ['java', 'groovy', 'kotlin'],
-                buildTools: config.buildTools ?: ['gradle', 'maven'],
-                testFramework: config.testFramework,
-                skipGradleTests: config.skipGradleTests ?: false,
-                skipMavenTests: config.skipMavenTests ?: false,
-                minimumJavaVersion: config.minimumJavaVersion,
-                maximumJavaVersion: config.maximumJavaVersion,
-                skips: (config.skips.collect { new GuideMetadata.Skip(it[0], it[1]) } ?: []).toSet(),
-                zipIncludes: config.zipIncludes ?: [],
-                env: config.env ?: [:],
-                apps: config.apps.collect { it ->
-                    new App(
-                            validateLicense: it.validateLicense == null ? true : it.validateLicense,
-                            framework: it.framework,
-                            testFramework: it.testFramework?.toUpperCase(),
-                            name: it.name,
-                            visibleFeatures: it.features ?: [],
-                            invisibleFeatures: it.invisibleFeatures ?: [],
-                            javaFeatures: it.javaFeatures ?: [],
-                            kotlinFeatures: it.kotlinFeatures ?: [],
-                            groovyFeatures: it.groovyFeatures ?: [],
-                            applicationType: it.applicationType ? ApplicationType.valueOf(it.applicationType.toUpperCase()) : ApplicationType.DEFAULT,
-                            excludeSource: it.excludeSource,
-                            excludeTest: it.excludeTest,
-                    )
-                }
+        Guide raw = jsonMapper.readValue(content, Guide.class)
+
+        return Optional.ofNullable(new Guide(
```

Comment:
@sdelamo I know creating a new Guide object is not ideal, but I was not sure how to do for a fields `slug` since a record is not mutable